feat: add maintenance window management to suppress alerts during planned activities

R3D2 opened the pull request:
• Implement timezone-aware scheduling with Daily, Weekly, and OneTime maintenance window types
• Create dedicated Maintenance tab in server configuration with DataTable-based management interface

<img width="1414" alt="Screenshot 2025-05-26 at 18 49 57" src="https://github.com/user-attachments/assets/8851a13c-5ae4-48d7-8303-cf96718c63c0" />
<img width="1445" alt="Screenshot 2025-05-26 at 18 50 08" src="https://github.com/user-attachments/assets/62915171-2e49-4391-9bd1-0897fb337c37" />
<img width="743" alt="Screenshot 2025-05-26 at 18 50 15" src="https://github.com/user-attachments/assets/809b55a4-a485-4046-9829-a556dfeea2f4" />

mbecker20 commented:
Thanks for this awesome feature! My only feedback is that its not clear enough in the UI what this Management Window does, a note describing that this disables alerting during this window would fix this.

Also, can you rebase this to the `1.18.0` branch? Then it is easy to include in the next release.

R3D2 commented:
Rebased the PR to 1.18.1 and added the wanted description.

<img width="1223" alt="Screenshot 2025-06-07 at 18 52 56" src="https://github.com/user-attachments/assets/397ce97d-0c67-4a71-893e-81899f12d9c5" />

R3D2 commented:
Rebased the PR to 1.18.1 and added the wanted description
<img width="1223" alt="Screenshot 2025-06-07 at 18 52 56" src="https://github.com/user-attachments/assets/748303bf-1111-46c1-8569-5520f5a4be60" />

mbecker20 commented:
Awesome, thanks! I'm just going to point it to 1.18.2 (currently even with main so I can do) to include in the next release. I will get that PR going shortly.

mbecker20 reviewed (approved):
I had to make some commits to fix some things in the rebase, but it looks good now, will merge
